pkg/util/grpcconnection/config.go

```diff
@@ -157,12 +152,27 @@ func createGRPCConnection(path string, timeout time.Duration, opts ...grpc.DialO
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
-	// grpc.DialContext is deprecated. TODO: Perhaps need to reconsider the approach in a future PR
-	//nolint:staticcheck
-	cc, err := grpc.DialContext(ctx, path, opts...)
+	cc, err := grpc.NewClient(path, opts...)
```

Comment:
```suggestion
	conn, err = grpc.NewClient(path, opts...)
```
We can use the variable defined in return types, which makes the defer function clear about which error it referring to.

Reply:
I tried it and encountered an issue: if there is an error, conn will be set to nil, resulting in `nil dereference` when trying to use `conn.Close`. Therefore, I changed it back, which also follows the implementation in [here](https://github.com/grpc/grpc-go/blob/f35fb347b7a7b57b92d25ca2e012dd8359b2a305/clientconn.go#L225-L241). 
Please take a look. Thank you!

Reply:
> I tried it and encountered an issue: if there is an error, conn will be set to nil, resulting in nil dereference when trying to use conn.Close in defer function.

Nice finding! Seems you tested the abnormal case, thanks.